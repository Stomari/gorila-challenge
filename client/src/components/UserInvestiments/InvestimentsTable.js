import React from 'react';

const InvestimentsTable = ({ investimentInfo, deleteInvestiment }) => {
  return (
    <table>
      <thead>
        <tr>
          <th>Data da compra</th>
          <th>Valor</th>
        </tr>
      </thead>
      <tbody>
        {
          investimentInfo && investimentInfo.map((element) => {
            return (
              <tr key={element._id}>
                <td>{element.date}</td>
                <td>{element.value}</td>
                <td><img className="delete" onClick={() => deleteInvestiment(element._id)} src="./images/trash.png" alt="Trash Icon" /></td>
              </tr>
            )
          })
        }
      </tbody>
    </table>
  )
}

export default InvestimentsTable;