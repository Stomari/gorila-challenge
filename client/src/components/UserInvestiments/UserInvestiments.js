import React from 'react';

const UserInvestiments = ({ fixedIncome, variableIncome, deleteInvestiment }) => {
  return (
    <section className="investiments">
      <div>
        <h2>Renda Fixa:</h2>
        <table>
          <thead>
            <tr>
              <th>Data da compra</th>
              <th>Valor</th>
            </tr>
          </thead>
          <tbody>
          {
            fixedIncome && fixedIncome.map((element) => {
              return (
                <tr key={element._id}>
                  <td>{element.date}</td>
                  <td>{element.value}</td>
                  <td><button onClick={() => deleteInvestiment(element._id)}>Delete</button></td>
                </tr>
                )
            })
          }
          </tbody>

        </table>
      </div>
      <div>
        <h2>Renda Vari√°vel:</h2>
        <table>
          <thead>
            <tr>
              <th>Data da compra</th>
              <th>Valor</th>
            </tr>
          </thead>
          <tbody>
          {
            variableIncome && variableIncome.map((element) => {
              return (
              <tr key={element._id}>
                <td>{element.date}</td>
                <td>{element.value}</td>
                <td><button onClick={() => deleteInvestiment(element._id)}>Delete</button></td>
              </tr>
              )
            })
          }
          </tbody>
        </table>
      </div>
    </section>
  )
}

export default UserInvestiments;