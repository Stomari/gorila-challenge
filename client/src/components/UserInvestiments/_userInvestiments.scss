.investiments {
  @include displayLocal(center);
  align-items: center;
  flex-direction: column;
  margin: 40px 0 100px 0;

  .invest-table {
    width: 900px;
    border: 2px solid black;
    .table-toggle {
      cursor: pointer;
      border-bottom: 2px solid black;
      padding: 20px 0 0 20px;
      h2 {
        margin-top: 0;
      }
    }
    .fixed-title {
      background-color: $bg-blue;
    }
    .variable-title {
      background-color: $bg-green;
    }
  }

  table {
    border-collapse: collapse;
    th, td {
      border: 1px solid black;
      padding: 10px 10px 10px 10px;
      font-size: 1.2rem;
    }
    th:nth-child(1) {
      width: 70%;
    }
    th:nth-child(2) {
      width: 25%;
    }
    td {
      text-align: center;
    }
    .delete {
      width: 30px;
      cursor: pointer;
    }
  }

}

@media (max-width: 1000px) {
  .investiments {
    .invest-table {
      width: 90vw;
    }
  }
}
