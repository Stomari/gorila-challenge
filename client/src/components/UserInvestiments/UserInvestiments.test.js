import React from 'react';
import { shallow } from 'enzyme';
import UserInvestiments from './UserInvestiments';

describe('UserInvestiments rendering', () => {
  const component = shallow(<UserInvestiments />); 

  it('should render', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render investiments', () => {
    expect(component.find('.investiments').length).toEqual(1);
  });

  it('should render fixed and variable investiments table', () => {
    expect(component.find('.invest-table').length).toEqual(2);
  });
});