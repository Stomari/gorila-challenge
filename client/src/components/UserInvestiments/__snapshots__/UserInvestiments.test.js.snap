// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInvestiments rendering should render 1`] = `ShallowWrapper {}`;
